### ISSUE TITLE
[Messenger] Performances

### ISSUE THREAD

**Issue by steevanb** [symfony/symfony#32513]
**Symfony version(s) affected**: tested on 4.3, probably same behavior on older versions.

**Description**  
When Messenger serialize a message, there is a huge performance difference if an EntityRepository is instantiated before the serialization or after.

If EntityRepository is not loaded before the serialization: ~ 21ms.

If EntityRepository is loaded before the serialization: ~ 2ms.

**How to reproduce**  
Everything is explained here, with an example and a Docker container to see it:

https://github.com/wizaplace/messenger-component-performance-test

**Comment by Tobion** [comment]
> When Messenger serialize a message

The message you dispatch is not serialized at all because you use sync message handling. I don't see the relation here.

**Comment by steevanb** [comment]
I was on this way cause if i change PhpSerializer to SymfonySerializer in configuration, i lost 15 ms.
I think you're right, but i don't understand why i have this behavior with SymfonySerializer.

**Comment by Tobion** [comment]
So can we close the issue or can you clarify the issue?

**Comment by steevanb** [comment]
I don't think you can close it. There is a real problem: see the performance difference between the 2 behaviors.
Don't close it just because i've said it is on serialization ;) If you prefer, it's when i send a message.

**Comment by wizacedric** [comment]
Dispatching a message is 10x faster (30ms -> 3ms) when an EntityRepository is injected into the sender, for not apparent reason. Sounds concerning.

**Comment by Tobion** [comment]
Are you running it in prod env? Can you run blackfire or any other tool to see why doing less takes more time?

**Comment by wizacedric** [comment]
> Are you running it in prod env?

I tried it, it's the same as dev env.

> Can you run blackfire or any other tool to see why doing less takes more time?

Not at the moment.

**Comment by xabbuh** [comment]
@wizacedric Do you have any news for us (a Blackfire profile, an example application or something like that)?

**Comment by steevanb** [comment]
See the "How to reproduce" section of my first message: you will find an application with everything to reproduce it ;)

**Comment by nicolas-grekas** [comment]
Anyone willing to provide a Blackfire profile+comparison? That'd be the way to move forward here.

**Comment by steevanb** [comment]
I've added BlackFire into Docker container: https://github.com/wizaplace/messenger-component-performance-test

You just need to clone the repository, add your BlackFire credentials, then launch `./docker/start.sh` to see the performances and the BlackFire profile.

**Comment by Nyholm** [comment]
The original github repository is gone. 
I spent some time trying to reproduce this but I failed. (I used SF ^5.0)

If I inject a doctrine repository in the class that runs `$bus->dispatch($message)`, it is a little slower, but not significantly. 

Here is the [baseline](https://blackfire.io/profiles/ea343a66-18f3-4d0b-b7de-487a22b70093/graph), the one [with repository](https://blackfire.io/profiles/26cbddf5-7b7d-471a-a354-21021b48c97a/graph) and a [comparison](https://blackfire.io/profiles/compare/17b4e315-8765-4480-afa6-96629922c701/graph)

I may have setup this all wrong, but without a way to reproduce this it is impossible to fix a potential issue.
